### PR TITLE
DataGrid: Implement fixed columns with sticky position

### PR DESCRIPTION
**Pull request by Alyar666**

### CODE REVIEW

File: packages/devextreme/js/localization/messages/en.json [hunk 1]
```diff
@@ -155,6 +155,7 @@
         "dxDataGrid-columnFixingUnfix": "Unfix",
         "dxDataGrid-columnFixingLeftPosition": "To the left",
         "dxDataGrid-columnFixingRightPosition": "To the right",
+        "dxDataGrid-columnFixingStickyPosition": "To the sticky",
```
Comment (vladaskorohodova) [inline packages/devextreme/js/localization/messages/en.json]:
```suggestion
        "dxDataGrid-columnFixingStickyPosition": "Stick in place",
```

Please redo all localization files.

File: packages/devextreme/js/__internal/grids/grid_core/sticky_columns/m_sticky_columns.ts [hunk 2]
```diff
@@ -69,7 +69,70 @@ const baseStickyColumns = <T extends ModuleType<ColumnsView>>(Base: T) => class
...
+                text: columnFixingOptions.texts.rightPosition, value: 'right', disabled: column.fixed && column.fixedPosition === 'right', onItemClick,
+              }],
+          },
+          {
+            text: columnFixingOptions.texts.unfix, value: 'none', disabled: !column.fixed, onItemClick,
+          },
+        );
+
+        if (this._isStickyColumns()) {
+          items[items.length - 2].items.push({
```
Comment (Raushen) [inline packages/devextreme/js/__internal/grids/grid_core/sticky_columns/m_sticky_columns.ts]: It is unclear code and unclear structure. I'd suggest to create items first and then to add them. For example:
```
const fixOptions = [{ ... }, { ... }];
if(this._isStickyColumns()) {
   fixOptions.push { ... value: 'sticky' ... }
}
items.push(fixOptions);
items.push(unfixOptions);
```

**Review by Raushen — DISMISSED** [review]

File: packages/devextreme/js/__internal/grids/grid_core/columns_controller/m_columns_controller.ts [hunk 3]
```diff
@@ -501,6 +502,17 @@ export class ColumnsController extends modules.Controller {
     return 0;
   }
 
+  public getStickyColumns(): any[] {
+    const visibleColumns = this.getVisibleColumns(null, true);
+    const isColumnFixing = this._isColumnFixing();
```
Comment (Raushen) [inline packages/devextreme/js/__internal/grids/grid_core/columns_controller/m_columns_controller.ts]: The result is not related to this._isColumnFixing() method. It seems it should return anyway:
`visibleColumns.filter((column) => column.fixed);`

File: packages/devextreme/js/__internal/grids/grid_core/columns_controller/m_columns_controller.ts [hunk 4]
```diff
@@ -742,7 +754,7 @@ export class ColumnsController extends modules.Controller {
           column.fixed = parentBandColumns[0]?.fixed ?? column.fixed;
           column.fixedPosition = parentBandColumns[0]?.fixedPosition ?? column.fixedPosition;
 
-          if (column.fixed) {
+          if (column.fixed && getFixedPosition(this, column) !== StickyPosition.Sticky) {
```
Comment (Raushen) [inline packages/devextreme/js/__internal/grids/grid_core/columns_controller/m_columns_controller.ts]: It seems the result is not related to the getFixedPosition method:
`if (column.fixed && column.fixedPosition !== StickyPosition.Sticky) {`

File: packages/devextreme/js/__internal/grids/grid_core/sticky_columns/utils.ts [hunk 5]
```diff
@@ -1,43 +1,110 @@
-import { StickyPosition } from './const';
+import { STICKY_BORDER_WIDTH, StickyPosition } from './const';
 
-const getStickyOffsetCore = function (columns, widths: number[], columnIndex): string {
-  let result = 0;
+export const getColumnFixedPosition = (
+  { fixedPosition }: { fixedPosition: StickyPosition | undefined },
+): StickyPosition => fixedPosition ?? StickyPosition.Left;
+
+const getColumnWidth = (column): number => (column.visibleWidth ?? column.width) as number;
```
Comment (Raushen) [inline packages/devextreme/js/__internal/grids/grid_core/sticky_columns/utils.ts:7]: Question: did you check if the column width can not be set by text: "200px" or "20%"?

**Review by vladaskorohodova — APPROVED** [review]

**Review by Raushen — APPROVED** [review]

**Review by mpreyskurantov — APPROVED** [review]

**Pull request merged**